dask_cuda/is_spillable_object.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import absolute_import, division, print_function
+
+from typing import Optional
+
+from dask.utils import Dispatch
+
+is_spillable_object = Dispatch(name="is_spillable_object")
+
+
+@is_spillable_object.register(list)
+@is_spillable_object.register(tuple)
+@is_spillable_object.register(set)
+@is_spillable_object.register(frozenset)
+def _(seq):
+    return any([is_spillable_object(s) for s in seq])
+
+
+@is_spillable_object.register(dict)
+def _(seq):
+    return any([is_spillable_object(s) for s in seq.items()])
+
+
+@is_spillable_object.register(object)
+def _(o):
+    return False
+
+
+@is_spillable_object.register_lazy("cudf")
+def register_cudf():
+    import cudf
+    from cudf.core.frame import Frame
+
+    @is_spillable_object.register(Frame)
+    def is_device_object_cudf_dataframe(df):
+        return cudf_spilling_status()
+
+    @is_spillable_object.register(cudf.BaseIndex)
+    def is_device_object_cudf_index(s):
+        return cudf_spilling_status()
+
+
+def cudf_spilling_status() -> Optional[bool]:
+    """Check the status of cudf's build-in spilling
+
+    Returns:
+        - True if cudf's internal spilling is enabled, or
+        - False if it is disabled, or
+        - None if the current version of cudf doesn't support spilling, or
```

Comment:
Since we don't depend on `cudf`, I guess people can have any version?

Reply:
This is theoretically right, but it isn't supported anyway. I guess it would make more sense for us to handle [cuDF as an optional dependency](https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#optional-dependencies) and then we could define the minimum version and drop checks like this, but we can address this on a follow-up PR.